### PR TITLE
Feature: Save and restore tabs when restart the computer with Files running

### LINKED ISSUE #12988: Feature: Save and restore tabs when restart the computer with Files running.
### What feature or improvement do you think would benefit Files?

I raised issue#11949 before.
This problem still exists today, directly shutting down or restarting the computer does not trigger Window_Closed() or AppUnhandledException(). Therefore, SaveSessionTabs() will not be executed.
I implemented this logic through other methods(https://github.com/IzayoiSake/Files), and the implementation method is as follows:
1.Add a new List CurrentTabsList in GeneralSettingsService.
2.When Tab changes, write Tabs containing ID information into SharedMemory and CurrentTabsList.
3.If ContinueLastSessionOnStartUp is checked, it will detect whether SharedMemory matches CurrentTabsList when starting the Files program.
4.If it does not match, it means that the computer restarted or the Files program was unexpectedly closed without triggering SaveSessionTabs().
5.Restore Tabs recorded in CurrentTabsList.
After running it on my computer, no abnormalities have been found for the time being. I want to pull a request for the code, but I am not sure whether it needs approval as written in ACKNOWLEDGE.
Affected functions include: OnNavigatedTo() and SaveSessionTabs() of Class MainPageViewModel.
Added some variables and functions (especially in MainPageViewModel).

### Requirements

-a approval to pull a request

### Files Version

2.5.10.0

### Windows Version

10.0.22621.1992

### Comments

_No response_

### PR DESCRIPTION
**Pull request by IzayoiSake**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #12988 
#11949
#5993

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?
   1. Open app ...
   2. Click settings button ...

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

File: src/Files.App/App.xaml.cs [hunk 1]
```diff
@@ -356,6 +357,8 @@ public static void SaveSessionTabs()
 				}
 			})
 			.ToList();
+			MainPageViewModel mainPageViewModel = Ioc.Default.GetRequiredService<MainPageViewModel>();
+			mainPageViewModel.RemoveThisInstanceTabs();
```
Comment (yaira2) [inline src/Files.App/App.xaml.cs:361]: App.xaml.cs isn't supposed to access mainPageViewModel

Comment (IzayoiSake) [reply src/Files.App/App.xaml.cs]: I changed it. App.xaml.cs now will directly access the SharedMemory and won't access MainPageViewModel.

File: src/Files.App/App.xaml.cs [hunk 2]
```diff
@@ -369,6 +383,105 @@ public static void SaveSessionTabs()
 				}
 			})
 			.ToList();
+			RemoveThisInstanceTabs();
+		}
+
+
+
+		public static MemoryMappedFile CheckSharedMemory()
```
Comment (0x5bfa) [inline src/Files.App/App.xaml.cs:391]: Can you move those static mothods into Helpers/AppLifecycle.cs you are going to create so we wont see any mess?

**Review by 0x5bfa — COMMENTED** [review]
I'm not familiar with memory stuff so I cannot approve.

**Pull request closed without merging**